classes/clang.bbclass

```diff
@@ -66,14 +67,18 @@ LDFLAGS:toolchain-clang:class-nativesdk = "${BUILDSDK_LDFLAGS} \
                                            -Wl,-rpath,${libdir}/.. "
 
 # Enable lld globally"
-LDFLAGS:append:toolchain-clang = "${@bb.utils.contains('DISTRO_FEATURES', 'ld-is-lld', ' -fuse-ld=lld', '', d)}"
+GOLDLINKER = "${@bb.utils.contains('DISTRO_FEATURES', 'ld-is-gold', 'gold', 'bfd', d)}"
+LINKER = "${@bb.utils.contains('DISTRO_FEATURES', 'ld-is-lld', 'lld', '${GOLDLINKER}', d)}"
+LDFLAGS:append:toolchain-clang = " -fuse-ld=${LINKER}"
 
 # Remove gcc specific -fcanon-prefix-map option, added in gcc-13+
 # clang does not support it yet
 DEBUG_PREFIX_MAP:remove:toolchain-clang = "-fcanon-prefix-map"
 
 # choose between 'gcc' 'clang' an empty '' can be used as well
 TOOLCHAIN ??= "gcc"
+# choose one of bfd, gold, lld ( may be 'mold' too )
+LINKER ??= "bfd"
```

Comment:
I think this has no effect because LINKER is previous assigned on line 71

Reply:
right I have taken care of it in v2. Thanks for reviewing.